src/main/java/com/faforever/client/fx/ImageViewHelper.java

```diff
@@ -0,0 +1,63 @@
+package com.faforever.client.fx;
+
+import javafx.beans.InvalidationListener;
+import javafx.beans.Observable;
+import javafx.beans.WeakInvalidationListener;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+public class ImageViewHelper {
+
+  public static void setDefaultImage(ImageView imageView, Image defaultImage) {
+    setDefaultImage(imageView, defaultImage, false);
+  }
+
+  public static void setDefaultImage(ImageView imageView, Image defaultImage, boolean onlyOnce) {
+    new ImageListenerImpl().setDefaultImage(imageView, defaultImage, onlyOnce);
+  }
+
+  private static class ImageListenerImpl {
+
+    private ImageView imageView;
+    private Image defaultImage;
+    private boolean onlyOnce;
+
+    private final InvalidationListener progressListener = new InvalidationListener() {
+      @Override
+      public void invalidated(Observable observable) {
+        Image image = imageView.getImage();
+        if (image != null && image.getProgress() == 1.0) {
+          if (image.isError()) {
+            imageView.setImage(defaultImage);
+          }
+          JavaFxUtil.removeListener(image.progressProperty(), this);
+        }
+      }
+    };
+
+    private final InvalidationListener imageListener = new InvalidationListener() {
+      @Override
+      public void invalidated(Observable observable) {
+        Image image = imageView.getImage();
+        if (image != null && image.getUrl() != null && !image.getUrl().startsWith("file:")) {
+          if (image.getProgress() == 1.0) {
+            if (image.isError()) {
+              imageView.setImage(defaultImage);
+            }
+          } else {
+            JavaFxUtil.addListener(image.progressProperty(), progressListener);
+          }
+        } else if (onlyOnce) {
+          JavaFxUtil.removeListener(imageView.imageProperty(), this);
+        }
+      }
+    };
+
+    public void setDefaultImage(ImageView imageView, Image defaultImage, boolean onlyOnce) {
+      this.imageView = imageView;
+      this.defaultImage = defaultImage;
+      this.onlyOnce = onlyOnce;
+      JavaFxUtil.addListener(imageView.imageProperty(), onlyOnce ? imageListener : new WeakInvalidationListener(imageListener));
```

Comment:
I don't quite understand the only once parameter. If the listener is always  removed when completed why do we need to specify a weak listener?

Reply:
Also wouldn't we want to addAndTrigger all these listeners?

Reply:
And might be nice if possible if the imageViews started with the default image maybe.

Reply:
I did a small refactoring of the class. Please review the code again.

Reply:
> Also wouldn't we want to addAndTrigger all these listeners?

No need. It is extra.

Reply:
> And might be nice if possible if the imageViews started with the default image maybe.

Done for some image views.